src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentLoaderService.cs

```diff
@@ -33,8 +35,15 @@ public PdbSourceDocumentLoaderService()
             // Otherwise, check the easiest (but most unlikely) case which is the document exists on the disk
             if (File.Exists(sourceDocument.FilePath))
             {
-                // TODO: Make sure the hash of the file is correct: https://github.com/dotnet/roslyn/issues/57351
-                return Task.FromResult<TextLoader?>(new FileTextLoader(sourceDocument.FilePath, Encoding.UTF8));
+                using var fileStream = new FileStream(sourceDocument.FilePath, FileMode.Open, FileAccess.Read, FileShare.Read | FileShare.Delete);
```

Comment:
>FileStream [](http://example.com/codeflow?start=43&length=10)

May throw.

Reply:
+1, wrap in https://github.com/dotnet/roslyn/blob/2ff43addcf69c5cbf2e3756daf551976e9d464b6/src/Workspaces/Core/Portable/Shared/Utilities/IOUtilities.cs#L25